src/pages/LandingPage/LandingPage.jsx

```diff
@@ -1,15 +1,39 @@
-import React from 'react';
+import React, { useState } from 'react'; // <-- useState is a react hook and imported here, you must always import react hooks
 import scrollToElement from '../../shared/utility/scrollToElement';
 import PortfolioCard from './PortfolioCard';
 import ContactForm from './ContactForm';
 import Render3DShape from './Render3DShape';
 
+const POSSIBLE_SHAPES = [
+  'torus',
+  'sphere',
+  'dodecahedron',
+  'icosahedron',
+  'octahedron',
+  'tetrahedron',
+  'torusknot',
+  'sahilsknot'
+];
+
+const INITIAL_SHAPE_INDEX = 0;
+const INITIAL_SHAPE = POSSIBLE_SHAPES[INITIAL_SHAPE_INDEX];
+
 const LandingPage = () => {
   const heroImage = 'hero.jpg';
+  const [shape, setShape] = useState(INITIAL_SHAPE);
+  let prevShapeIndex = INITIAL_SHAPE_INDEX;
 
+  const handleClick = () => {
+    let randomIndex = prevShapeIndex;
+    while (randomIndex === prevShapeIndex) {
+      randomIndex = Math.floor(Math.random() * POSSIBLE_SHAPES.length);
+    }
+    prevShapeIndex = randomIndex;
+    setShape(POSSIBLE_SHAPES[randomIndex]);
+  };
   return (
     <div>
-      <Render3DShape className="fixed top-0 left-0 z-0 w-screen h-screen opacity-50" />
+      <Render3DShape className="fixed top-0 left-0 z-0 w-screen h-screen opacity-50" shape={shape} />
```

Comment:
smart!

Reply:
If you dont know already: Render3DShape will rerender whenever shape changes, this is good, cause then the new shape will be shown on the screen.